### PR TITLE
Add initial set of composition puzzles

### PR DESCRIPTION
**Pull request by ablev-nv**
usda files and explanatory notes on some common composition issues, excerpted from a soon-to-be publicly available presentation on USD composition.

### CODE REVIEW

File: test_assets/CompositionPuzzles/PayloadAndReference/README.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
+# Payload and reference ordering
+
+![problem](images/Slide108.PNG)
+
+This puzzle is inspired by a real-life USD asset structure problem at encountered during USD adoption at a large animation studio. 
+
+In the studio's production pipeline, before animation begins on a shot, the shot populated with non-deforming models, placed on the stage for blocking.
+
+shot.usda has sublayers of animation.usda and layout.usda, and layout.usda
```
Comment (AlexSchwank) [inline test_assets/CompositionPuzzles/PayloadAndReference/README.md:9]: Use `shot.usda` syntax for file names to make it easier to read?

Comment (ablev-nv) [reply test_assets/CompositionPuzzles/PayloadAndReference/README.md:9]: Ah good call!

Comment (ablev-nv) [reply test_assets/CompositionPuzzles/PayloadAndReference/README.md]: Done.

File: test_assets/CompositionPuzzles/PayloadAndReference/README.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
+# Payload and reference ordering
+
+![problem](images/Slide108.PNG)
+
+This puzzle is inspired by a real-life USD asset structure problem at encountered during USD adoption at a large animation studio.
```
Comment (beersandrew) [inline test_assets/CompositionPuzzles/PayloadAndReference/README.md:5]: small grammar suggestion:

This puzzle was inspired by a real-life USD asset structure problem encountered during USD adoption at a large animation studio.

File: test_assets/CompositionPuzzles/PayloadAndReference/README.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
+# Payload and reference ordering
+
+![problem](images/Slide108.PNG)
+
+This puzzle is inspired by a real-life USD asset structure problem at encountered during USD adoption at a large animation studio. 
+
+In the studio's production pipeline, before animation begins on a shot, the shot populated with non-deforming models, placed on the stage for blocking.
+
+`shot.usda` has sublayers of `animation.usda` and `layout.usda`, and `layout.usda`
```
Comment (beersandrew) [inline test_assets/CompositionPuzzles/PayloadAndReference/README.md:9]: small grammar suggestion:

`shot.usda` has sublayers of `animation.usda` and `layout.usda`. `layout.usda` 
has a reference to `model.usda`, the non-deforming model geometry.

File: test_assets/CompositionPuzzles/PayloadAndReference/README.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+has a reference to `model.usda`, the non-deforming model geometry.
+
+![problem 2](images/Slide109.PNG)
+
+Animation performs their work at a later date and delivers their animation caches by adding a payload to their sublayer. But when this happened, they didn't see the deforming geometry from animation on the stage.
```
Comment (beersandrew) [inline test_assets/CompositionPuzzles/PayloadAndReference/README.md:14]: small grammar suggestion:

Animation performed their work at a later date and delivered their animation caches by adding a payload to their sublayer. When this happened, they didn't see the deforming geometry from animation on the stage.

**Review by beersandrew — COMMENTED** [review]
This was cool to go through, and very helpful for someone new to USD. Added some small comments about grammar I noticed while reading, feel free to take or ignore them as you like

**Review by meshula — APPROVED** [review]

**Comment by meshula** [comment]
@ablev-nv Ready to land I think?

**Comment by ablev-nv** [comment]
Let's do it. Thanks @meshula

**Pull request merged**